src/coreclr/jit/block.h

```diff
@@ -697,8 +697,49 @@ struct BasicBlock : private LIR::Range
         bbJumpEhf  = jumpEhf;
     }
 
+private:
     BasicBlockFlags bbFlags;
 
+public:
+    BasicBlockFlags CheckFlag(const BasicBlockFlags flag) const
+    {
+        // Assert flag is not multiple BasicBlockFlags OR'd together
+        // by checking if it is a power of 2
+        // (CheckFlag expects to check only one flag at a time)
+        assert((flag & (flag - 1)) == 0);
+        return (bbFlags & flag);
+    }
+
+    BasicBlockFlags CheckFlags(const BasicBlockFlags flags) const
+    {
```

Comment:
Should you assert `flags` is not a power of two here?

Reply:
I think that makes sense.